### PR TITLE
APIMF-2453: added ReferenceCollector to always collect in an ordered fashion

### PR DESCRIPTION
**Pull request by tomsfernandez**

### CODE REVIEW

File: shared/src/main/scala/amf/client/render/RenderOptions.scala [hunk 1]
```diff
@@ -14,6 +14,7 @@ import scala.scalajs.js.annotation.{JSExportAll, JSExportTopLevel}
 class RenderOptions {
 
   private var sources: Boolean       = false
+  private var rawSourceMaps: Boolean = false
```
Comment (AgustinBettati) [inline shared/src/main/scala/amf/client/render/RenderOptions.scala]: Unrelated change

Comment (tomsfernandez) [reply shared/src/main/scala/amf/client/render/RenderOptions.scala]: Removed

**Review by AgustinBettati — APPROVED** [review]
